tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -331,12 +371,16 @@ public static void main(String[] args) {
     } while (!am.singleNode); // We don't retry on single node training.
     // Wait for the worker nodes to finish (The interval between registering the exit code to final exit)
     am.stop();
+    completed = System.currentTimeMillis();
     am.printTaskUrls();
     if (exitOnError) {
       LOG.fatal("Error running TonyApplicationMaster !!");
       System.exit(-1);
     }
 
+    // By this time jobDir should have been set
+    createHistoryFile(fs, createMetadataObj(started, completed, succeeded), jobDir);
```

Comment:
Are we still trying to write to this file if this call failed?

Reply:
I'd say no, but I'm not sure how to alert users when this method fails because they might check the history server and not see their job displayed. Maybe I can just have it retry creating the file for a certain number of times?

Reply:
👍